controllers/volumereplication_controller.go

```diff
@@ -167,15 +167,12 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		}
 	} else {
 		if contains(instance.GetFinalizers(), volumeReplicationFinalizer) {
-			// Disable replication only if it is in primary state and PV object
-			// is present.
-			if instance.Spec.ReplicationState == replicationv1alpha1.Primary && pvErr == nil {
-				err := r.disableVolumeReplication(logger, volumeHandle, parameters, secret)
-				if err != nil {
-					logger.Error(err, "failed to disable replication")
-					return ctrl.Result{}, err
-				}
+			err := r.disableVolumeReplication(logger, volumeHandle, parameters, secret)
```

Comment:
We need to call disable without checking state of the VR in the case of Delete as well, which currently calls disable only if it is Primary, right?

Reply:
Yes correct I will fix it as part of https://github.com/csi-addons/volume-replication-operator/pull/104

Reply:
sorry, discard my last comment. This is reverting back the change and there is no check on the VR state to call the disableReplication. Am I missing something?

Reply:
We should call disable unconditionally [here](https://github.com/csi-addons/volume-replication-operator/blob/00f42351bccb8020ef90d36e65d4320959154285/controllers/volumereplication_controller.go#L172-L173) as well, right? As the storage provider needs to handle if it is `Primary` and do the appropriate action.

Reply:
isn't it doing at the same place?

Reply:
yes, ack! sorry wires crossed between PR reviews.